target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
Thanks a lot for this detailed explanation. If this special workaround is kept in the end, it should be part of the commit message!

It is possible to create a gzipped kernel without the uImage header:

```
define Device/RS353
  DEVICE_PROFILE := RS353
  DEVICE_TITLE := Bintec RS353
  KERNEL := kernel-bin | append-dtb | gzip
  IMAGE_SIZE := 31232k
  DEVICE_PACKAGES := kmod-usb-dwc2 kmod-ath9k wpad-mini kmod-i2c-gpio kmod-swconfig wpad-mini kmod-ledtrig-usbdev
endef
TARGET_DEVICES += RS353
```

The Boss Image is prefixed with a 0x34 byte long header. I'm quite sure, this header contains the image length and the target crc32. So far I wasn't able to spot a field that defines the kernel entry address. You can create a custom build step which adds this header to your image as well. Since you have control over the length, you can try to limit the checksum calculation to the kernel only or to kernel + rootfs (before the pad-rootfs step):

```
define Device/RS353
  DEVICE_PROFILE := RS353
  DEVICE_TITLE := Bintec RS353
  KERNEL := kernel-bin | append-dtb | gzip
  IMAGE_SIZE := 31232k
  IMAGE/sysupgrade.bin := append-kernel | prefix_kernel_with_custom_header_build_step | append-rootfs | pad-rootfs
  DEVICE_PACKAGES := kmod-usb-dwc2 kmod-ath9k wpad-mini kmod-i2c-gpio kmod-swconfig wpad-mini kmod-ledtrig-usbdev
endef
TARGET_DEVICES += RS353
```

But the kernel entry address of 0x2d6b0 could be a problem. In LEDE (the OpenWrt and LEDE code will be synced soon) we do have already some tools which might be the key to success here. For ar71xx we have a [relocate-kernel](https://github.com/lede-project/source/blob/master/target/linux/ar71xx/image/Makefile#L79) build step, which sound like something that could do the job.

What might even better work is the [lzma loader](https://github.com/openwrt/openwrt/blob/master/target/linux/lantiq/image/Makefile#L21). Since we have only limited flash space, we want to use the best compression for the kernel. Since your board isn't the first where the bootloader that isn't aware of lzma, the lzma loader was written. The lzma loader is put in front of the kernel, is run by the bootloader, uncompresses the lzma compressed kernel and jumps into the kernel.

If you drop the uImage header and use the boss header to the image you need to write a custom [mtd-splitter](https://github.com/openwrt/openwrt/tree/master/target/linux/generic/files/drivers/mtd/mtdsplit) (to create the kernel, rootfs and rootfs_data from the firmware partition during runtime). 

I'm still not sure in which endianess mode the SoC runs (The SoC can run in little and big endian). Your flash dump contains for example mtd0.bin and mtd0s.bin. U-Boot can be compiled for little endian as well as big endian. It's just that one endianess is the default and at least the bootloader on your board seem to use the other endianess (and for lantiq code the not matching endianess is the - not fixed - default). What might be possible as well, that the bootloader and the boss image are using different endianesses.

What do you think about replacing the u-boot in favour of a "standard" OpenWrt solution?